composables/transaction/mintToken/transactionMintStatemine.ts

```diff
@@ -10,11 +10,9 @@ type id = { id: number }
 export const assignIds = <T extends TokenToMint>(tokens: T[]): (T & id)[] => {
   let lastId = 0
   return tokens.map((token) => {
-    const { alreadyMinted, lastIndexUsed } =
-      token.selectedCollection as MintedCollection
+    const { lastIndexUsed } = token.selectedCollection as MintedCollection
 
-    // Use the maximum value between lastId and alreadyMinted or lastIndexUsed
-    lastId = Math.max(lastIndexUsed, alreadyMinted, lastId)
+    lastId = Math.max(lastIndexUsed, lastId)
```

Comment:
viki collection had the following nft ids : [01,2]
which means
`lastIndexUsed = 2`
`alreadyMinted = 3`
`lastId = 3`

and that was the problem, lastId should have been 2

Reply:
Please consider using array reduce on this

Reply:
Please consider using array reduce here

Reply:
```
return tokens.reduce(({ result, lastId }, token) => {
  const { lastIndexUsed } = token.selectedCollection as MintedCollection;
  const newLastId = Math.max(lastIndexUsed, lastId) + 1;

  return {
    result: [...result, { ...token, id: newLastId }],
    lastId: newLastId
  };

}, { result: [], lastId: 0 });
```

I was thinking of removing the lastId initialization

Reply:
```
return tokens.reduce(({ result, lastId }, token) => {
  const { lastIndexUsed } = token.selectedCollection as MintedCollection;
  const newLastId = Math.max(lastIndexUsed, lastId) + 1;

  return {
    result: [...result, { ...token, id: newLastId }],
    lastId: newLastId
  };

}, { result: [], lastId: 0 });
```

I was thinking of removing the lastId initialization

It's just like it's better to use map instead of having an initializer and using for of loop